### ISSUE TITLE
False positive "Cannot move"

### ISSUE THREAD

**Issue by KizzyCode** [intellij-rust/intellij-rust#3062]
## Environment

* **Intellij-Rust plugin version: 0.2.0.2110-182**
* **Rust toolchain version: stable-x86_64-apple-darwin - rustc 1.30.1 (1433507eb 2018-11-07)**
* **IDE name and version: IntelliJ IDEA 2018.2.6 (Ultimate Edition)**
* **Operating system: macOS Mojave 10.14.1**

## Problem description
There is a false positive if you want to dereference a `Copy` type in a generic trait implementation that takes `self` by reference. However since `Self` implements `Copy` this is perfectly valid and compiles fine.

## Steps to reproduce
```rust
trait Dummy where Self: Copy {
    fn dummy(&self) -> Self {
        *self // ← This shows the error "Cannot move" ("Inspection info: Implements simple cases of borrow checking.")
    }
}
```

**Comment by vlad20012** [comment]
@ortem Let's blacklist Self type for move analysis for now.
